packages/playback-core/src/index.ts

```diff
@@ -392,6 +409,36 @@ export const loadMedia = (
           );
         }
       });
+    } else if (props.autoplay) {
+      hls.once(Hls.Events.MANIFEST_PARSED, () => {
+        const oldMuted = mediaEl.muted;
+        const restoreMuted = () => (mediaEl.muted = oldMuted);
+        switch (props.autoplay) {
+          // ANY:
+          // try to play with current options
+          // if it fails, mute and try playing again
+          // if that fails, restore muted state and don't try playing again
+          case AutoplayTypes.ANY:
+            mediaEl.play().catch((error: Error) => {
+              mediaEl.muted = true;
+              mediaEl.play().catch(restoreMuted);
+            });
+            break;
+          // MUTED:
+          // mute the player and then try playing
+          // if that fails, restore muted state
+          case AutoplayTypes.MUTED:
+            mediaEl.muted = true;
+            mediaEl.play().catch(restoreMuted);
+
+            break;
+          // Default or if autoplay is a boolean attribute:
+          // Try playing the video and catch the failed autoplay warning
+          default:
```

Comment:
nit (non-blocking): Since we're not guaranteed that folks are using typescript and since TS doesn't provide runtime type checking, this means we'd treat any truthy autoplay value that isn't `"any"` or `"muted"` as `autoplay=true`. Probably not a huge deal, but just calling it out.

Reply:
I think that would be correct behavior anyway? Since it's a boolean attribute, and we want to maintain backwards compatibility here.

Reply:
these are instance properties, not attributes.... you may still be right, just want to make sure the two aren't being conflated.

Reply:
ah, good call. Another thing to check tomorrow, then.